server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -913,6 +915,7 @@ protected Node(
                 xContentRegistry,
                 recoverySettings
             );
+            CryptoManagerRegistry.initRegistry(pluginsService.filterPlugins(CryptoPlugin.class));
```

Comment:
Why is this static, as opposed to creating an instance and passing it to where it needs to go? Keeping static state like this is generally at odds with good objected oriented design.

Reply:
I know that it is against the design but there is no way to pass it to a BlobContainer/BlobStore instance. We can't change abstractions as they are blob specific abstractions and we need a way to create cryptoManager which can encrypt on all transfers.

Reply:
Moved it to  a single instance pattern.